Purchases/Purchasing/StoreKitAbstractions/PromotionalOffer.swift

```diff
@@ -33,6 +33,43 @@ extension PromotionalOffer {
                                  signature: self.signature,
                                  timestamp: self.timestamp as NSNumber)
     }
+
+    /// - Throws: ``ErrorCode/unexpectedBackendResponseError`` if the signature cannot be encoded in UTF8
+    @available(iOS 15.0, tvOS 15.0, watchOS 8.0, macOS 12.0, *)
+    var sk2PurchaseOption: Product.PurchaseOption {
+        get throws {
+            guard let signatureData = self.signature.data(using: .utf8) else {
+                throw ErrorUtils.unexpectedBackendResponse(
+                    withSubError: Self.Error.invalidSignature(self.signature)
+                )
+            }
```

Comment:
👏

Reply:
maybe we can log just before this throw?

Reply:
Errors have the location from which they were created in their `userInfo` since #1078.
I agree that the SDK should probably always log errors before returning them to the user for visibility, but I feel like that should happen at the top layer instead of in every call site. We have 
`ErrorUtils.logErrorIfNeeded`, perhaps we should make sure that all of the entry points call that?

Reply:
agreed!!

Reply:
yeah, trying to consolidate where to throw errors from has been a challenge in the past. I love the idea of just always doing it at the top level

Reply:
We already have https://app.shortcut.com/revenuecat/story/10723/ios-refactor-logging-of-errors, I added a note there pointing to this.